Civi/Api4/Generic/Traits/ArrayQueryActionTrait.php

```diff
@@ -149,6 +149,12 @@ private function filterCompare($row, $condition) {
         $pattern = '/^' . str_replace('%', '.*', preg_quote($expected, '/')) . '$/i';
         return !preg_match($pattern, $value) == ($operator != 'LIKE');
 
+      case 'REGEXP':
+      case 'NOT REGEXP':
+        // MySQL uses case-insensitive REGEXP matching by default, so we replicate that here
+        $pattern = '/' . str_replace('/', '\/', $expected) . '/i';
```

Comment:
use `preg_quote()` instead of `str_replace()`

Reply:
The issue with that is that we actually want the supplied value to be interpreted as a PCRE pattern rather than a quoted string.  if `$expected` has a value of `(a|b)`, we want `$pattern` to look like `/(a|b)/i`; with `preg_quote()`, we'd get `/\(a\|b\)/i`.

There's a bit of a security concern with accepting arbitrary regular expressions [as it could lead to DoS](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS), but I'm pretty sure you could do the same thing by building an API4 query that keeps the database busy with silly joins or similar, so I'm not too worried about that.

Reply:
Ok, got it.